core/src/main/java/org/apache/accumulo/core/manager/balancer/TServerStatusImpl.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.manager.balancer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.accumulo.core.master.thrift.TabletServerStatus;
+import org.apache.accumulo.core.spi.balancer.data.TServerStatus;
+import org.apache.accumulo.core.spi.balancer.data.TableStatistics;
+
+public class TServerStatusImpl implements TServerStatus {
+  private final TabletServerStatus thriftStatus;
+  private Map<String,TableStatistics> tableInfoMap;
+
+  public static TServerStatusImpl fromThrift(TabletServerStatus tss) {
+    return (tss == null) ? null : new TServerStatusImpl(tss);
+  }
+
+  public TServerStatusImpl(TabletServerStatus thriftStatus) {
+    Objects.requireNonNull(thriftStatus);
+
+    this.thriftStatus = thriftStatus;
```

Comment:
This is cleanest with a static import of `requireNonNull`:
```suggestion
    this.thriftStatus = requireNonNull(thriftStatus);
```
But, even without a static import, you can still one-line this:
```suggestion
    this.thriftStatus = Objects.requireNonNull(thriftStatus);
```

Reply:
Thanks, updated for the next commit.